src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.Expressions.cs

```diff
@@ -307,25 +307,26 @@ private void EmitGetArrayElement(Type arrayType)
             if (arrayType.IsSZArray)
             {
                 // For one dimensional arrays, emit load
-                _ilg.EmitLoadElement(arrayType.GetElementType());
+                _ilg.EmitLoadElement(arrayType.GetElementType()!);
             }
             else
             {
                 // Multidimensional arrays, call get
-                _ilg.Emit(OpCodes.Call, arrayType.GetMethod("Get", BindingFlags.Public | BindingFlags.Instance));
+                _ilg.Emit(OpCodes.Call, arrayType.GetMethod("Get", BindingFlags.Public | BindingFlags.Instance)!);
             }
         }
 
-        private void EmitSetIndexCall(IndexExpression node, Type objectType)
+        private void EmitSetIndexCall(IndexExpression node, Type? objectType)
         {
             if (node.Indexer != null)
             {
                 // For indexed properties, just call the setter
-                MethodInfo method = node.Indexer.GetSetMethod(nonPublic: true);
+                MethodInfo method = node.Indexer.GetSetMethod(nonPublic: true)!;
                 EmitCall(objectType, method);
             }
             else
             {
+                Debug.Assert(objectType != null);
```

Comment:
>Debug.Assert(objectType != null); [](start = 16, length = 33)

`EmitGetIndexCall` uses `!` rather than `Debug.Assert()`. Consider changing one of the two methods so the methods are consistent.